### PR TITLE
Build user groups

### PR DESCRIPTION
**Pull request by justinschw**
Adding a build user's authority groups to the variables.

After we know who is running the build, we need to validate that they have permission, but the apis to do so are protected in jenkins, so we would like the plugin to populate that data as well.

### CODE REVIEW

**Comment by martinda** [comment]
Can you please add the relevant unit test(s) as well?

**Comment by justinschw** [comment]
> Can you please add the relevant unit test(s) as well?

I went ahead and added the unit tests. However, since the code is very old, in order to get the unit tests to compile and run, I had to update many things in the pom.xml as well as some jelly files. Please review these changes.

**Comment by martinda** [comment]
Hello @justinschw , thanks for adding the tests. I made a few updates in a PR I sent directly to your fork https://github.com/justinschw/build-user-vars-plugin/pull/1

I do not actively develop plugins, so I do not know if I missed anything important, but overall it looks good to me.

**Comment by justinschw** [comment]
> Hello @justinschw , thanks for adding the tests. I made a few updates in a PR I sent directly to your fork [justinschw#1](https://github.com/justinschw/build-user-vars-plugin/pull/1)
> 
> I do not actively develop plugins, so I do not know if I missed anything important, but overall it looks good to me.

Thanks for the input. Is there anyone I can contact to do a review who is able to merge?

**Comment by fabiodcasilva** [comment]
> > Hello @justinschw , thanks for adding the tests. I made a few updates in a PR I sent directly to your fork [justinschw#1](https://github.com/justinschw/build-user-vars-plugin/pull/1)
> > I do not actively develop plugins, so I do not know if I missed anything important, but overall it looks good to me.
> 
> Thanks for the input. Is there anyone I can contact to do a review who is able to merge?

@justinschw I can do the merge. Can you integrate the changes that @martinda did on the branch so that everything is verified? Thanks a lot for your contribution.

**Comment by justinschw** [comment]
> > > Hello @justinschw , thanks for adding the tests. I made a few updates in a PR I sent directly to your fork [justinschw#1](https://github.com/justinschw/build-user-vars-plugin/pull/1)
> > > I do not actively develop plugins, so I do not know if I missed anything important, but overall it looks good to me.
> > 
> > 
> > Thanks for the input. Is there anyone I can contact to do a review who is able to merge?
> 
> @justinschw I can do the merge. Can you integrate the changes that @martinda did on the branch so that everything is verified? Thanks a lot for your contribution.

Thanks Fabio. I have merged the review as requested.

**Comment by martinda** [comment]
@justinschw Hummm... I see you have merged my commits to your master branch, but it should have been merged to `justinschw:build_user_groups` (my mistake for not sending them to the right place). I think you need to put my commits to this PR. Your master is probably diverging from this repo too. Sorry I think I messed up a bit.

**Comment by justinschw** [comment]
Hi martinda,
I moved all of your changes and mine to the build_user_groups branch, and I reset my master to this commit:

d3a18f08491e06b3713511359b09610904648794

So now all of the changes that I intended to make to this plugin, as well as the changes you made, are now in this branch and part of the pull request. Our master branches should be in sync.

**Comment by martinda** [comment]
Thanks @justinschw , I can locally merge the PR to master.

**Comment by justinschw** [comment]
The failures appear to be with a jenkins configuration.

File: README.md [hunk 1]
```diff
@@ -37,4 +38,4 @@ node {
 
 ## Jenkins core compatibility
 
-This plugin requires Jenkins 1.609+ (since 1.5 version of plugin).
+This plugin requires Jenkins 2.164.1+ (since 1.8 version of plugin).
```
Comment (fabiodcasilva) [inline README.md:41]: The next release will be 1.7. can you just point to 1.7 or just remove the information in parentheses?

**Review by fabiodcasilva — APPROVED** [review]

**Comment by TobiX** [comment]
Would it be okay to make the dependency on the SAML plugin optional? (I can provide a pull request, if needed)

**Comment by justinschw** [comment]
Would you have to build it manually? Or would you just be notified that if you make the selection then you need the saml plugin as well?

If it were something simple as the latter, then it would be fine.

⁣Get BlueMail for Android ​

**Comment by fabiodcasilva** [comment]
@TobiX what will happen if you choose not to have the dependency? To access the BUILD_USER_GROUPS variable the dependency is needed.

**Comment by TobiX** [comment]
> @TobiX what will happen if you choose not to have the dependency? To access the BUILD_USER_GROUPS variable the dependency is needed.

It will probably be empty. No loss for anybody who wasn't using saml before, since fetching groups from an unconfigured authentication plugin won't work anyways...

**Comment by fabiodcasilva** [comment]
> > @TobiX what will happen if you choose not to have the dependency? To access the BUILD_USER_GROUPS variable the dependency is needed.
> 
> It will probably be empty. No loss for anybody who wasn't using saml before, since fetching groups from an unconfigured authentication plugin won't work anyways...

therefore I don't see any issue. @justinschw is it ok for you?
@TobiX can you create the pull request, please?

**Pull request merged**